src/subdomains/generic/user/models/auth/auth-lnurl.service.ts

```diff
@@ -0,0 +1,143 @@
+import { ForbiddenException, Injectable, NotFoundException } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { secp256k1 } from '@noble/curves/secp256k1';
+import { randomBytes } from 'crypto';
+import { Config, Process } from 'src/config/config';
+import { LightningHelper } from 'src/integration/lightning/lightning-helper';
+import { IpLogService } from 'src/shared/models/ip-log/ip-log.service';
+import { DfxLogger } from 'src/shared/services/dfx-logger';
+import { Lock } from 'src/shared/utils/lock';
+import { Util } from 'src/shared/utils/util';
+import { AuthService } from 'src/subdomains/generic/user/models/auth/auth.service';
+import {
+  AuthLnurlResponseDto,
+  AuthLnurlResponseStatus,
+  AuthLnurlSignupDto,
+} from 'src/subdomains/generic/user/models/auth/dto/auth-lnurl.dto';
+
+export interface AuthCacheDto {
+  k1: string;
+  k1CreationTime: number;
+  accessToken?: string;
+  accessTokenCreationTime?: number;
+}
+
+@Injectable()
+export class AuthLnUrlService {
+  private readonly logger = new DfxLogger(AuthLnUrlService);
+
+  private authCache: Map<string, AuthCacheDto> = new Map();
+
+  constructor(private readonly authService: AuthService, private readonly ipLogService: IpLogService) {}
+
+  @Cron(CronExpression.EVERY_30_SECONDS)
+  @Lock()
+  processCleanupAccessToken() {
+    if (Config.processDisabled(Process.LNURL_AUTH_CACHE)) return;
+
+    const before30SecTime = Util.secondsBefore(30).getTime();
+
+    const entriesToBeUpdated = [...this.authCache.entries()].filter(
+      (k) => k[1].accessTokenCreationTime < before30SecTime,
+    );
+
+    entriesToBeUpdated.forEach((e) => {
+      e[1].accessToken = undefined;
+      e[1].accessTokenCreationTime = undefined;
+    });
+  }
+
+  @Cron(CronExpression.EVERY_5_MINUTES)
+  @Lock()
+  processCleanupAuthCache() {
+    if (Config.processDisabled(Process.LNURL_AUTH_CACHE)) return;
+
+    const before5MinTime = Util.minutesBefore(5).getTime();
+
+    const keysToBeDeleted = [...this.authCache.entries()]
+      .filter((k) => k[1].k1CreationTime < before5MinTime)
+      .map((k) => k[0]);

```

Comment:
Same here?

Reply:
Here I filter over the values and need the corresponding keys.
How does this work, if I just iterate over the values?

Reply:
Yes, makes sense. You could use the k1 stored in the value but it's probably cleaner to use the key directly.